packages/react-ui-kit/src/Misc/useTimeout.ts

```diff
@@ -17,21 +17,21 @@
  *
  */
 
-import {useEffect, useState} from 'react';
+import {useEffect, useRef} from 'react';
 
 const useTimeout = (onTimeoutStop: () => void, duration: number) => {
-  const [timerId, setTimerId] = useState(null);
+  const timerRef = useRef<number | null>(null);
 
   const removeTimeout = () => {
-    if (timerId) {
-      window.clearTimeout(timerId);
-      setTimerId(null);
+    if (timerRef) {
```

Comment:
Should be timerRef.current

Reply:
Won't timerRef always be defined since it's object?

Reply:
As far as I know they check for deep equality. So if current changes it should be recognized. But we can be more precise here and go for .current like you mentioned